### PR TITLE
Refactor authentication related code.

### PR DESCRIPTION
**Pull request by robrap**
Before merge:
- [ ] squash
- [ ] adjust commit comment to mention ARCH-244

### CODE REVIEW

File: edx_rest_framework_extensions/middleware.py [hunk 1]
```diff
@@ -1,76 +1,19 @@
...
-    def process_view(self, request, view_func, view_args, view_kwargs):  # pylint: disable=unused-argument
-        # Views as functions store the view's class in the 'view_class' attribute.
-        # Viewsets store the view's class in the 'cls' attribute.
-        view_class = getattr(
-            view_func,
-            'view_class',
-            getattr(view_func, 'cls', view_func),
-        )
+from rest_framework_jwt.authentication import (
+    BaseJSONWebTokenAuthentication,
```
Comment (robrap) [inline edx_rest_framework_extensions/middleware.py]: @nasthagiri: This is probably backward compatibility overkill, but means less of a chance of something going wrong.  It was just used in one of our classes, but someone could have imported it.  Should I leave this or kill it?

File: edx_rest_framework_extensions/middleware.py [hunk 2]
```diff
@@ -1,76 +1,19 @@
...
+)  # pylint: disable=unused-import
+from edx_rest_framework_extensions.auth.jwt.middleware import (
+    EnsureJWTAuthSettingsMiddleware,
+    JwtAuthCookieMiddleware,
+)  # pylint: disable=unused-import
 
-        view_authentication_classes = getattr(view_class, 'authentication_classes', tuple())
-        if self._includes_base_class(view_authentication_classes, BaseJSONWebTokenAuthentication):
-            self._add_missing_jwt_permission_classes(view_class)
+from .permissions import NotJwtRestrictedApplication   # pylint: disable=unused-import
```
Comment (robrap) [inline edx_rest_framework_extensions/middleware.py]: @nasthagiri: Same question about leaving or killing.  I'd hope we didn't import from here, and that we'd be ok with cleaning up an IDA if it did?

File: edx_rest_framework_extensions/middleware.py (hunk 1 above)
Comment (robrap) [reply edx_rest_framework_extensions/middleware.py]: FYI: I'm leaning toward removing these, but want you to weigh-in.

File: edx_rest_framework_extensions/middleware.py (hunk 2 above)
Comment (robrap) [reply edx_rest_framework_extensions/middleware.py]: FYI: I'm leaning toward removing these, but want you to weigh-in.

File: edx_rest_framework_extensions/middleware.py (hunk 1 above)
Comment (nasthagiri) [reply edx_rest_framework_extensions/middleware.py]: Sure. Just leave it. We already have a TODO here - so we can remove all BWC code at once.

File: edx_rest_framework_extensions/middleware.py (hunk 2 above)
Comment (nasthagiri) [reply edx_rest_framework_extensions/middleware.py]: see above.

**Pull request merged**